### ISSUE TITLE
v0.5.1 sometimes random apps open on menu select

### ISSUE THREAD

**Issue by 01micko** [labwc/labwc#308]
This is pre scene-graph only and I should have reported it sooner, however I don't know if/how long you want to support the v0.5.x series.

When opening a menu entry _sometimes_ a random second app opens. Video attached, first take.

https://user-images.githubusercontent.com/1019119/165451347-59297e62-f693-44f4-ae4f-903831c012d8.mp4

( :rofl: **no** I'm not plugging vivladi :laughing: )

**Comment by Consolatis** [comment]
Would you mind posting your `menu.xml` (or some fresh one which shows the same issue)?

**Comment by 01micko** [comment]
Using a menu generator and custom icon font with this, but I don't see that as the cause, works perfectly in latest master.

```xml
<openbox_menu>
<menu id="xdg-menu-Desktop" label=" Desktop">
<item label=" Desktop settings">
<action name="Execute"><command>pdesktop</command></action>
</item>
<item label=" Font manager">
<action name="Execute"><command>fontmanager</command></action>
</item>
<item label=" pSynclient Touchpad Setup">
<action name="Execute"><command>psynclient</command></action>
</item>
<item label=" Wallpaper Setter">
<action name="Execute"><command>wallpaper</command></action>
</item>
<item label=" pTheme">
<action name="Execute"><command>ptheme</command></action>
</item>
<item label=" Chooselocale">
<action name="Execute"><command>chooselocale</command></action>
</item>
<item label=" Psync">
<action name="Execute"><command>psync</command></action>
</item>
<item label=" Screensaver setup">
<action name="Execute"><command>pupx</command></action>
</item>
<item label=" Set date and time">
<action name="Execute"><command>set-time-for-puppy</command></action>
</item>
<item label=" Set timezone">
<action name="Execute"><command>timezone-set</command></action>
</item>
<item label=" Font viewer">
<action name="Execute"><command>gfontsel</command></action>
</item>
</menu>
<menu id="xdg-menu-System" label=" System">
<item label=" Boot manager">
<action name="Execute"><command>bootmanager</command></action>
</item>
<item label=" CPU Frequency Scaling Tool">
<action name="Execute"><command>wcpufreq</command></action>
</item>
<item label=" Foot">
<action name="Execute"><command>foot</command></action>
</item>
<item label=" Foot Server">
<action name="Execute"><command>foot --server</command></action>
</item>
<item label=" IPInfo">
<action name="Execute"><command>ipinfo</command></action>
</item>
<item label=" Labwc Config">
<action name="Execute"><command>labwc-tweaks</command></action>
</item>
<item label=" Pup-Kview">
<action name="Execute"><command>Pup-Kview</command></action>
</item>
<item label=" Pup-SysInfo">
<action name="Execute"><command>Pup-SysInfo</command></action>
</item>
<item label=" Sys-Info System Information">
<action name="Execute"><command>sys-info</command></action>
</item>
<item label=" Task Manager">
<action name="Execute"><command>lxtask</command></action>
</item>
<item label=" Video-Info Graphics Information">
<action name="Execute"><command>video-info</command></action>
</item>
<item label=" Yambar">
<action name="Execute"><command>yambar</command></action>
</item>
<item label=" dconf Editor">
<action name="Execute"><command>dconf-editor</command></action>
</item>
<item label=" Event manager">
<action name="Execute"><command>eventmanager</command></action>
</item>
<item label=" Htop">
<action name="Execute"><command>/usr/bin/urxvt -e htop</command></action>
</item>
<item label=" Yambar">
<action name="Execute"><command>yambar</command></action>
</item>
<item label=" pSchedule">
<action name="Execute"><command>/usr/local/bin/pschedule</command></action>
</item>
<item label=" GParted">
<action name="Execute"><command>gparted_shell</command></action>
</item>
<item label=" Login manager">
<action name="Execute"><command>loginmanager</command></action>
</item>
</menu>
<menu id="xdg-menu-Setup" label=" Setup">
<item label=" Dependency checker">
<action name="Execute"><command>/usr/local/petget/check_deps_gui.sh</command></action>
</item>
<item label=" Package Manager NG">
<action name="Execute"><command>gpkgdialog</command></action>
</item>
<item label=" Package Manager NG (cli)">
<action name="Execute"><command>defaultterminal -e sh -c 'defaultterminal -e pkgdialog'</command></action>
</item>
<item label=" Package manager">
<action name="Execute"><command>/usr/local/petget/pkg_chooser.sh</command></action>
</item>
<item label=" Puppy setup">
<action name="Execute"><command>wizardwizard</command></action>
</item>
<item label=" QuickSetup">
<action name="Execute"><command>quicksetup</command></action>
</item>
<item label=" SFS-Load">
<action name="Execute"><command>sfs_load</command></action>
</item>
<item label=" Set default apps">
<action name="Execute"><command>puppyapps</command></action>
</item>
<item label=" CUPS print manager">
<action name="Execute"><command>cups_shell</command></action>
</item>
<item label=" Download Vivaldi Browser">
<action name="Execute"><command>install_vivaldi_gui.sh</command></action>
</item>
<item label=" Internet connection wizard">
<action name="Execute"><command>connectwizard</command></action>
</item>
<item label=" Menu manager">
<action name="Execute"><command>menumanager</command></action>
</item>
<item label=" Mouse/Keyboard manager">
<action name="Execute"><command>input-wizard</command></action>
</item>
<item label=" Screen/Graphics manager">
<action name="Execute"><command>xorgwizard</command></action>
</item>
<item label=" DiskPup - minimal frugal installer">
<action name="Execute"><command>diskpup</command></action>
</item>
<item label=" FrugalPup - flexible frugal installer">
<action name="Execute"><command>frugalpup</command></action>
</item>
<item label=" Grub4Dos bootloader config">
<action name="Execute"><command>grub4dosconfig</command></action>
</item>
<item label=" Monitor Settings">
<action name="Execute"><command>lxrandr</command></action>
</item>
<item label=" Puppy installer">
<action name="Execute"><command>/usr/sbin/dotpup1</command></action>
</item>
<item label=" Remaster Puppy">
<action name="Execute"><command>/usr/sbin/remasterpup2</command></action>
</item>
<item label=" Remove builtin packages">
<action name="Execute"><command>/usr/sbin/remove_builtin</command></action>
</item>
<item label=" StickPup - usb stick installer">
<action name="Execute"><command>stickpup</command></action>
</item>
<item label=" f2StickPup - f2fs usb stick installer">
<action name="Execute"><command>f2stickpup</command></action>
</item>
</menu>
<menu id="xdg-menu-Utility" label=" Utility">
<item label=" Graphical Disk Map">
<action name="Execute"><command>gdmap</command></action>
</item>
<item label=" UExtract">
<action name="Execute"><command>/usr/local/apps/UExtract/AppRun</command></action>
</item>
<item label=" Xarchiver">
<action name="Execute"><command>xarchiver   </command></action>
</item>
<item label=" Xdelta">
<action name="Execute"><command>xdelta_gui</command></action>
</item>
<item label=" Xffm Patch manager">
<action name="Execute"><command>xfdiff4</command></action>
</item>
<item label=" pupMd5sum Graphical Checksum">
<action name="Execute"><command>pupmd5sum.sh</command></action>
</item>
<item label=" MoManager">
<action name="Execute"><command>momanager</command></action>
</item>
<item label=" Pudd">
<action name="Execute"><command>Pudd</command></action>
</item>
<item label=" Pupsave Backup">
<action name="Execute"><command>pupsave-backup</command></action>
</item>
<item label=" Resize personal storage">
<action name="Execute"><command>resizepfile.sh</command></action>
</item>
<item label=" Save folder backup and restore">
<action name="Execute"><command>SaveFolderBackup</command></action>
</item>
<item label=" Snap2 snapshot backup">
<action name="Execute"><command>snap2</command></action>
</item>
<item label=" pMirror">
<action name="Execute"><command>pmirror</command></action>
</item>
<item label=" Foot">
<action name="Execute"><command>foot</command></action>
</item>
<item label=" Foot Server">
<action name="Execute"><command>foot --server</command></action>
</item>
<item label=" Gexec">
<action name="Execute"><command>gexec</command></action>
</item>
<item label=" LXTerminal">
<action name="Execute"><command>lxterminal</command></action>
</item>
<item label=" gmrun execute a command">
<action name="Execute"><command>gmrun</command></action>
</item>
</menu>
<menu id="xdg-menu-Filesystem" label=" Filesystem">
<item label=" Graphical Disk Map">
<action name="Execute"><command>gdmap</command></action>
</item>
<item label=" File Manager PCManFM">
<action name="Execute"><command>pcmanfm   </command></action>
</item>
<item label=" FileMnt">
<action name="Execute"><command>filemnt -manager</command></action>
</item>
<item label=" Pmount">
<action name="Execute"><command>pmount</command></action>
</item>
<item label=" PupMTP">
<action name="Execute"><command>pupmtp</command></action>
</item>
<item label=" pFind">
<action name="Execute"><command>pfind</command></action>
</item>
<item label=" Partview">
<action name="Execute"><command>partview</command></action>
</item>
</menu>
<menu id="xdg-menu-Graphic" label=" Graphic">
<item label=" GNU Image Manipulation Program">
<action name="Execute"><command>gimp-2.10</command></action>
</item>
<item label=" GNU Image Manipulation Program">
<action name="Execute"><command>gimp-2.10</command></action>
</item>
<item label=" mtPaint">
<action name="Execute"><command>mtpaint   </command></action>
</item>
<item label=" PupCamera">
<action name="Execute"><command>pupcamera</command></action>
</item>
<item label=" Color Picker">
<action name="Execute"><command>wl_colpick</command></action>
</item>
<item label=" Gcolor2">
<action name="Execute"><command>gcolor2</command></action>
</item>
<item label=" Take a Shot!">
<action name="Execute"><command>tas</command></action>
</item>
</menu>
<menu id="xdg-menu-Document" label=" Document">
<item label=" Get LibreOffice download and install LibreOffice">
<action name="Execute"><command>get_libreoffice</command></action>
</item>
<item label=" NoteCase notes manager">
<action name="Execute"><command>notecase</command></action>
</item>
<item label=" AbiWord">
<action name="Execute"><command>abiword</command></action>
</item>
<item label=" Geany">
<action name="Execute"><command>geany   </command></action>
</item>
<item label=" L3afpad">
<action name="Execute"><command>l3afpad   </command></action>
</item>
<item label=" Document Viewer">
<action name="Execute"><command>evince</command></action>
</item>
<item label=" Pdict">
<action name="Execute"><command>pdict</command></action>
</item>
<item label=" puppyPDF">
<action name="Execute"><command>puppypdf</command></action>
</item>
</menu>
<menu id="xdg-menu-Business" label=" Business">
<item label=" Gnumeric">
<action name="Execute"><command>gnumeric</command></action>
</item>
<item label=" Galculator">
<action name="Execute"><command>galculator</command></action>
</item>
<item label=" Gmeasures units converter">
<action name="Execute"><command>gmeasures</command></action>
</item>
</menu>
<menu id="xdg-menu-Personal" label=" Personal">
<item label=" DidiWiki personal wiki">
<action name="Execute"><command>didiwiki-gui</command></action>
</item>
<item label=" Figaro's Password Manager 2">
<action name="Execute"><command>fpm2</command></action>
</item>
<item label=" Osmo">
<action name="Execute"><command>osmo</command></action>
</item>
<item label=" Bcrypt">
<action name="Execute"><command>bcrypt_gui</command></action>
</item>
<item label=" Gnumeric">
<action name="Execute"><command>gnumeric</command></action>
</item>
</menu>
<menu id="xdg-menu-Network" label=" Network">
<item label=" Firewall Setup">
<action name="Execute"><command>firewall_ng</command></action>
</item>
<item label=" PcurlFtp file sharing">
<action name="Execute"><command>rox -d /root/File-Sharing</command></action>
</item>
<item label=" PureFTPd">
<action name="Execute"><command>run-pureftpd</command></action>
</item>
<item label=" Samba Simple Management">
<action name="Execute"><command>samba.sh</command></action>
</item>
<item label=" pGPRS">
<action name="Execute"><command>pgprs --connect</command></action>
</item>
</menu>
<menu id="xdg-menu-Internet" label=" Internet">
<item label=" Badwolf">
<action name="Execute"><command>badwolf</command></action>
</item>
<item label=" Firefox">
<action name="Execute"><command>run-as-spot firefox</command></action>
</item>
<item label=" Vivaldi">
<action name="Execute"><command>run-as-spot vivaldi-stable --enable-features=UseOzonePlatform --ozone-platform=wayland</command></action>
</item>
<item label=" Vivaldi (snapshot)">
<action name="Execute"><command>run-as-spot vivaldi-snapshot --enable-features=UseOzonePlatform --ozone-platform=wayland</command></action>
</item>
<item label=" IRC Chat Client (irssi)">
<action name="Execute"><command>xterm -e irssi-shell</command></action>
</item>
<item label=" Claws Mail">
<action name="Execute"><command>claws-mail</command></action>
</item>
<item label=" PMirrorget website downloader">
<action name="Execute"><command>pmwget</command></action>
</item>
<item label=" Transmission">
<action name="Execute"><command>transmission-gtk   </command></action>
</item>
<item label=" Pup-Advert-Blocker">
<action name="Execute"><command>pup-advert-blocker</command></action>
</item>
</menu>
<menu id="xdg-menu-Multimedia" label=" Multimedia">
<item label=" Celluloid">
<action name="Execute"><command>celluloid</command></action>
</item>
<item label=" FFConvert multimedia converter">
<action name="Execute"><command>ffconvert</command></action>
</item>
<item label=" mhWaveEdit audio recorder/editor">
<action name="Execute"><command>mhwaveedit</command></action>
</item>
<item label=" Precord mp3,ogg,wav recorder/player">
<action name="Execute"><command>/usr/bin/precord</command></action>
</item>
<item label=" PulseAudio Volume Control">
<action name="Execute"><command>pavucontrol</command></action>
</item>
<item label=" PupMixer">
<action name="Execute"><command>pupmixer</command></action>
</item>
<item label=" pMusic">
<action name="Execute"><command>pmusic</command></action>
</item>
<item label=" pMusic podcast">
<action name="Execute"><command>pmusic_podcast</command></action>
</item>
<item label=" Pavrecord Audio-Video recorder/player">
<action name="Execute"><command>pavrecord</command></action>
</item>
<item label=" ISO Master">
<action name="Execute"><command>isomaster</command></action>
</item>
<item label=" pBurn">
<action name="Execute"><command>pburn</command></action>
</item>
</menu>
<menu id="xdg-menu-Fun" label=" Fun">
<item label=" Rubix cube puzzle">
<action name="Execute"><command>rubix</command></action>
</item>
</menu>
<menu id="root-menu" label=" ">
<menu id="xdg-menu-Desktop" />
<menu id="xdg-menu-System" />
<menu id="xdg-menu-Setup" />
<item label="____________">
</item>
<menu id="xdg-menu-Utility" />
<menu id="xdg-menu-Filesystem" />
<menu id="xdg-menu-Graphic" />
<menu id="xdg-menu-Document" />
<menu id="xdg-menu-Business" />
<menu id="xdg-menu-Personal" />
<menu id="xdg-menu-Network" />
<menu id="xdg-menu-Internet" />
<menu id="xdg-menu-Multimedia" />
<menu id="xdg-menu-Fun" />
<item label="____________">
</item>
<item label=" Help"><action name="Execute"><command>puppyhelp</command></action>
</item>
<item label=" Leave"><action name="Execute"><command>logout_gui</command></action>
</item>
</menu>
</openbox_menu>
```

**Comment by 01micko** [comment]
> I don't know if/how long you want to support the v0.5.x series.

Might be better to spend devel time focusing on v0.6.0 rather than concentrate on this since it is imminent. #303

**Comment by Consolatis** [comment]
Assuming this is fixed on 0.6 / latest master branch. Feel free to comment if it still happens on these current versions.

**Comment by 01micko** [comment]
It is in 0.6

Thanks!
